packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+function maskDate(rawInput = '', valueOnly = false): any {
+  const RE_DATE = /^(\d{1,2})[\D]?(\d{1,2})?[\D]?(\d{1,4})?/;
+  const match = RE_DATE.exec(rawInput);
+  let formattedDate = '';
+  if (match) {
+    const [month, day, year] = match.slice(1);
+    formattedDate = [
+      // We treat all non-numeric characters as a delimiter. If they're using a
+      // delimiter after a month or day, we interpret that as the user supplying
+      // a single digit for month or day, which we will automatically pad for them.
+      month && month.padStart(2, '0'),
+      day && day.padStart(2, '0'),
+      year,
+    ]
+      .filter((s) => s)
+      .join('/');
+  }
+  if (valueOnly) {
+    return formattedDate;
+  }
+  const hint = 'MM/DD/YYYY';
+  const hintSub = hint.substring(formattedDate.length);
+  return formattedDate + hintSub;
+}
+
+export const BuiltInMask = {
+  DATE: maskDate,
+};
+
+export interface LabelMaskProps {
+  /**
+   * Must contain a `TextField` component
+   */
+  children: React.ReactNode;
+  labelMask?: (rawInput: string, valueOnly?: boolean) => string;
```

Comment:
We'll want to add a doc comment here to explain how the function is used and what the two props are about